src/test/java/com/mongodb/hibernate/jdbc/MongoStatementIntegrationTests.java

```diff
@@ -0,0 +1,349 @@
+    static void afterAll() {
+        if (sessionFactory != null) {
+            sessionFactory.close();
+        }
+    }
+
+    @BeforeEach
+    void setUp() {
+        session = sessionFactory.openSession();
+    }
+
+    @AfterEach
+    void tearDown() {
+        if (session != null) {
+            session.close();
+        }
+    }
+
+    private static final String COLLECTION_NAME = "books";
+
+    @Nested
+    class ExecuteUpdateTests {
+
+        @BeforeEach
+        void setUp() {
+            session.doWork(conn -> {
+                conn.createStatement()
+                        .executeUpdate(
+                                """
+                        {
+                            delete: "%s",
+                            deletes: [
+                                { q: {}, limit: 0 }
+                            ]
+                        }"""
+                                        .formatted(COLLECTION_NAME));
+            });
+        }
+
+        private static final String INSERT_MQL =
+                """
+                 {
+                            insert: "%s",
+                            documents: [
+                                {
+                                    _id: 1,
+                                    title: "War and Peace",
+                                    author: "Leo Tolstoy",
+                                    outOfStock: false
+                                },
+                                {
+                                    _id: 2,
+                                    title: "Anna Karenina",
+                                    author: "Leo Tolstoy",
+                                    outOfStock: false
+                                },
+                                {
+                                    _id: 3,
+                                    title: "Resurrection",
+                                    author: "Leo Tolstoy",
+                                    outOfStock: false
+                                },
+                                {
+                                    _id: 4,
+                                    title: "Crime and Punishment",
+                                    author: "Fyodor Dostoevsky",
+                                    outOfStock: false
+                                },
+                                {
+                                    _id: 5,
+                                    title: "The Brothers Karamazov",
+                                    author: "Fyodor Dostoevsky",
+                                    outOfStock: false
+                                },
+                                {
+                                    _id: 6,
+                                    title: "Fathers and Sons",
+                                    author: "Ivan Turgenev",
+                                    outOfStock: false
+                                }
+                            ]
+                        }"""
```

Comment:
As far as I can see, these tests can get by with three documents, two of which are authored by "Leo Tolstoy". Let's reduce the number of prepared documents, as that also reduces the number of expected documents, significantly reducing the total number of lines in the test class, which makes it easier to read.

Reply:
good point! Yeah, 3 docs suffice.